rewrite-yaml/src/main/java/org/openrewrite/yaml/FormatPreservingReader.java

```diff
@@ -79,7 +83,7 @@ public int read(@NonNull char[] cbuf, int off, int len) throws IOException {
             for (int i = 0; i < read; i++) {
                 char e = cbuf[i];
                 if (Character.UnicodeBlock.of(e) != Character.UnicodeBlock.BASIC_LATIN && i % 2 == 0) {
-                    bufferIndex--;
+                    unicodeOffset++;
```

Comment:
Wondering if we'd need to reset this counter when we read the next "chunk". But let's first see if we even keep this in the solution we end up with.